Fix audit order

jan-cerny opened the pull request:
#### Description:
Uses a better regular expression to check audit rule order in template_OVAL_audit_rules_unsuccessful_file_modification_rule_order.

The new proposed regular expression is `^first$\n(^(?!first|third).*$\n)*^second$\n(^(?!second|first).*$\n)*^third$` where:
* `first` is line with rule for creating file
* `second` is line with rule for write or truncate
* `third` is line with rule for any other unsuccessful file acces

#### Rationale:
The old regular expression didn't work properly and was very demanding on recursive matching. The old regular expression triggered a segmentation fault described in (https://bugzilla.redhat.com/show_bug.cgi?id=1686467)

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/ComplianceAsCode/content/inspections/201db144-d926-4e85-affb-0dc86285e130) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/ComplianceAsCode/content/inspections/201db144-d926-4e85-affb-0dc86285e130) completed: **1 new issues**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/ComplianceAsCode/content/inspections/a434198e-90f4-4029-8ed9-edf7c4e4ebcb) was created.

jan-cerny commented on shared/templates/template_OVAL_audit_rules_unsuccessful_file_modification_rule_order:
```diff
@@ -183,13 +183,25 @@
+      <!-- ^first$\n(^(?!first|third).*$\n)*^second$\n(^(?!second|first).*$\n)*^third$ -->
+      <literal_component>^</literal_component>
       <object_component item_field="text" object_ref="object_arufm_{{{ SYSCALL }}}_order_32bit_a20100_eacces_augenrules" />
-      <variable_component var_ref="var_audit_rule_{{{ SYSCALL }}}_order_separator_regex" />
```
done

jan-cerny commented:
I have removed the unused variable which holed the separator.

I have rebased the pull request on top of the master branch which contains #4064 .

On my Fedora machine I have copied https://github.com/linux-audit/audit-userspace/blob/master/rules/30-ospp-v42.rules to `/etc/audit/rules.d/` and the following rules from Fedora OSPP profile pass:
* audit_rules_unsuccessful_file_modification_open_by_handle_at_rule_order
* audit_rules_unsuccessful_file_modification_open_rule_order
* audit_rules_unsuccessful_file_modification_openat_rule_order

yuumasato commented:
Great work, thank you for making the expression more robust.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/ComplianceAsCode/content/inspections/a434198e-90f4-4029-8ed9-edf7c4e4ebcb) completed: **No new issues**
